legend-engine-xts-relationalStore/legend-engine-xt-relationalStore-execution/legend-engine-xt-relationalStore-executionPlan/src/main/java/org/finos/legend/engine/plan/execution/stores/relational/plugin/RelationalExecutionNodeExecutor.java

```diff
@@ -1981,6 +1985,23 @@ private DoubleStrategyHashMap<Object, Object, SQLExecutionResult> switchedParent
         return parentMap;
     }
 
+    private DoubleStrategyHashMap<Object, Object, SQLExecutionResult> switchedParentHashMapPerChildResult(RelationalGraphObjectsBatch relationalGraphObjectsBatch, int parentIndex, ResultSet childResultSet, Supplier<List<String>> parentPrimaryKeyColumnsSupplier, boolean isDatabaseIdentifiersCaseSensitive)
+    {
+        List<String> parentPrimaryKeyColumnNames = parentPrimaryKeyColumnsSupplier.get();
+        List<Integer> parentPrimaryKeyIndices;
+        if (!isDatabaseIdentifiersCaseSensitive)
+        {
+            parentPrimaryKeyIndices = parentPrimaryKeyColumnNames.stream().map(key -> key.toUpperCase()).map(FunctionHelper.unchecked(childResultSet::findColumn)).collect(Collectors.toList());
```

Comment:
Should it always be toUpperCase? Could we do something like this

```
Map<String,Integer> m = Maps.mutable.empty();
int columnCount = childResultSet.getMetaData().getColumnCount();
for (int i=1; i<=columnCount; i++) {
    m.put(childResultSet.getMetaData().getColumnName(i).toUpperCase(),i);
}
List<Integer> parentPrimaryKeyIndices = parentPrimaryKeyColumnsSupplier.get().stream().map(String::toUpperCase).map(m::get).collect(Collectors.toList());
```

Reply:
yes, it makes more sense.